src/command/repo/get.c

```diff
@@ -49,10 +68,78 @@ storageGetProcess(IoWrite *destination)
+                        if (cfgOptionStrNull(cfgOptStanza) != NULL && !strEq(stanza, cfgOptionStr(cfgOptStanza)))
+                        {
+                            THROW_FMT(
+                                OptionInvalidValueError, "stanza name '%s' given in option doesn't match the given path",
+                                strPtr(cfgOptionStr(cfgOptStanza)));
+                        }
+
+                        if (strEq(strLstGet(filePathSplitLst, 0), STORAGE_PATH_ARCHIVE_STR))
+                        {
+                            // Find the archive passphrase
+                            if (!strEndsWithZ(file, INFO_ARCHIVE_FILE) && !strEndsWithZ(file, INFO_ARCHIVE_FILE INFO_COPY_EXT))
+                            {
+                                InfoArchive *info = infoArchiveLoadFile(
+                                    storageRepo(), strNewFmt(STORAGE_PATH_ARCHIVE "/%s/%s", strPtr(stanza), INFO_ARCHIVE_FILE),
+                                    repoCipherType, cipherPass);
+                                cipherPass = infoArchiveCipherPass(info);
+                            }
+                        }
+                        else if (strEq(strLstGet(filePathSplitLst, 0), STORAGE_PATH_BACKUP_STR))
+                        {
+                            if (!strEndsWithZ(file, INFO_BACKUP_FILE) && !strEndsWithZ(file, INFO_BACKUP_FILE INFO_COPY_EXT))
+                            {
+                                // Find the backup passphrase
+                                InfoBackup *info = infoBackupLoadFile(
+                                    storageRepo(), strNewFmt(STORAGE_PATH_BACKUP "/%s/%s", strPtr(stanza), INFO_BACKUP_FILE),
+                                    repoCipherType, cipherPass);
+                                cipherPass = infoBackupCipherPass(info);
+
+                                // Find the manifest passphrase
+                                if (!strEq(strLstGet(filePathSplitLst, 2), STRDEF(BACKUP_PATH_HISTORY)) &&
+                                    !strEndsWithZ(file, BACKUP_MANIFEST_FILE) &&
+                                    !strEndsWithZ(file, BACKUP_MANIFEST_FILE INFO_COPY_EXT))
+                                {
+                                    const Manifest *manifest = manifestLoadFile(
+                                        storageRepo(), strNewFmt(STORAGE_PATH_BACKUP "/%s/%s/%s", strPtr(stanza),
+                                        strPtr(strLstGet(filePathSplitLst, 2)), BACKUP_MANIFEST_FILE), repoCipherType, cipherPass);
+                                    cipherPass = manifestCipherSubPass(manifest);
+                                }
+                            }
+                        }
+                        // Only archive and backup directories are allowed
+                        else
+                            THROW_FMT(OptionInvalidValueError, "unable to determine encryption key for '%s'", strPtr(file));
```

Comment:
I don't like that we have the same error message two times. There should be a way to combine them.  Perhaps set cipherPass = NULL if we can't find it and base the error message on that?

Reply:
That seems reasonable indeed. I pushed a small modification to do so.

Reply:
Looks good. That's pretty much what I was thinking.